File: Problems/0901.-1000./0941._Valid_Mountain_Array/ValidMountainArray2.py
#!/bin/python3
class Solution:
    def validMountainArray(self, arr: list) -> bool:
        i, j, k = 0, len(arr)-1, 0

        # Loop over the entire array
        while (k < len(arr)):
            if (i < len(arr)-1 and arr[i+1] > arr[i]): # Strictly INCREASING
                i += 1
            if (j > 0 and arr[j] < arr[j-1]): # Strictly DECREASING
                j -= 1
            k += 1
        return (i > 0 and i == j and j < len(arr)-1)

File: Problems/0901.-1000./0941._Valid_Mountain_Array/ValidMountainArray2.java
//!/bin/java
class Solution {
    public boolean validMountainArray(int[] arr) {
        int n = arr.length, i = 0, j = n-1, k = 0;

        // Loop over the entire array
        while (k < n) {
            if (i < n-1 && arr[i+1] > arr[i]) // Strictly INCREASING
                ++i;
            if (j > 0 && arr[j] < arr[j-1]) // Strictly DECREASING
                --j;
            k++;
        }
        return (i > 0 && i == j && j < n-1);
    }
}
